worker/uniter/runner/context/context.go

```diff
@@ -892,6 +981,17 @@ func (ctx *HookContext) Flush(process string, ctxErr error) (err error) {
 		}
 	}
 
+	// TODO: We need a uniter API to persist the following in a single txn:
+	// - the state
+	// - relation data
+	// - storage changes
+	// - podspec
+	if ctx.cacheValues != nil && writeChanges {
```

Comment:
There is no way to call code which will initialize cacheValues, so we're safe with this for now.

Reply:
I really like this approach as it allows us to implement the server-side bits in a different PR.